lib/rules/no-constant-condition.js

```diff
@@ -32,17 +34,20 @@ module.exports = function(context) {
                 return true;
 
             case "UnaryExpression":
-                return isConstant(node.argument);
+                return (node.operator === "typeof" && inBooleanPosition) ||
+                    isConstant(node.argument, true);
 
             case "BinaryExpression":
             case "LogicalExpression":
-                return isConstant(node.left) && isConstant(node.right) && node.operator !== "in";
+                return isConstant(node.left, false) &&
```

Comment:
LogicalExpressions should pass `inBooleanPosition` through. Other `BinaryExpreesion`s should not.